.github/workflows/mdbook_cicd.yml

```diff
@@ -43,9 +48,9 @@ jobs:
           name: spec_mdbook
           path: spec_mdbook.tar
 
-      #- name: Deploy mdBook to github pages
-      #   uses: peaceiris/actions-gh-pages@v3
-      #   if: ${{ github.ref == 'refs/heads/main' }}
-      #   with:
-      #     github_token: ${{ secrets.GITHUB_TOKEN }}
-      #     publish_dir: ./book
+      - name: Deploy mdBook to github pages 🚀
+        uses: JamesIves/github-pages-deploy-action@4.0.0
+        # check tag
+        if: startsWith(github.ref, 'refs/tags/')
+        with:
+          folder: ./book
```

Comment:
on:
  create:
will ensure only tags following dated pattern trigger this workflow

here the if statement will ensure, deployment will only work for tags and not direct push to main or PR based trigger